meta-sokol-flex-staging/recipes-kernel/lttng/lttng-tools/0001-Ensure-that-the-consumerd-configure-arguments-are-us.patch

```diff
@@ -1,17 +1,50 @@
-From 0d09ed08a84ec77e01bede24847f4ced4e99057c Mon Sep 17 00:00:00 2001
+From 62705486851068915cb507f20aa4b8665eacb8ef Mon Sep 17 00:00:00 2001
 From: Christopher Larson <chris_larson@mentor.com>
 Date: Fri, 29 Sep 2023 17:07:30 -0500
 Subject: [PATCH] Ensure that the consumerd configure arguments are used
 
-Currently, the --with-consumerd32-bin, --with-consumerd32-libdir, etc arguments
-are not being used by lttng-sessiond, which causes problems with providing a
-well configured multilib tracing environment. Ensure that they are used as the
-default values in sessiond.
+Currently, the --with-consumerd32-bin, --with-consumerd32-libdir, etc
+arguments are not being used by lttng-sessiond, which causes problems
+with providing a well configured multilib tracing environment. Ensure
+that they are used as the default values in sessiond. Also ensure that
+the paths are present before spawning them.
 
 Signed-off-by: Christopher Larson <chris_larson@mentor.com>
+
 ---
-Upstream-Status: Pending
+ src/bin/lttng-sessiond/client.c          | 3 +++
+ src/bin/lttng-sessiond/sessiond-config.c | 8 ++++----
+ src/common/defaults.h                    | 4 ++++
+ 3 files changed, 11 insertions(+), 4 deletions(-)
 
+diff --git a/src/bin/lttng-sessiond/client.c b/src/bin/lttng-sessiond/client.c
+index c798234..e972ca6 100644
+--- a/src/bin/lttng-sessiond/client.c
++++ b/src/bin/lttng-sessiond/client.c
+@@ -1002,6 +1002,7 @@ static int process_client_msg(struct command_ctx *cmd_ctx, int *sock,
+ 		int *sock_error)
+ {
+ 	int ret = LTTNG_OK;
++	struct stat st;
+ 	bool need_tracing_session = true;
+ 	bool need_domain;
+ 	bool need_consumerd;
+@@ -1285,6 +1286,7 @@ static int process_client_msg(struct command_ctx *cmd_ctx, int *sock,
+ 			/* 64-bit */
+ 			pthread_mutex_lock(&the_ustconsumer64_data.pid_mutex);
+ 			if (the_config.consumerd64_bin_path.value &&
++					stat(the_config.consumerd64_bin_path.value, &st) == 0 &&
+ 					the_ustconsumer64_data.pid == 0 &&
+ 					cmd_ctx->lsm.cmd_type != LTTNG_REGISTER_CONSUMER) {
+ 				pthread_mutex_unlock(&the_ustconsumer64_data.pid_mutex);
+@@ -1314,6 +1316,7 @@ static int process_client_msg(struct command_ctx *cmd_ctx, int *sock,
+ 			/* 32-bit */
+ 			pthread_mutex_lock(&the_ustconsumer32_data.pid_mutex);
+ 			if (the_config.consumerd32_bin_path.value &&
++					stat(the_config.consumerd32_bin_path.value, &st) == 0 &&
```

Comment:
when the code reaches this point, `the_config.consumerd32_bin_path.value` has been resolved to `/home/root/` so we still get the issue as this path exists. so maybe add a check here that it is a regular file, and not a directory?

Reply:
Good catch, thanks

Reply:
Hmm, when I build, I'm not seeing the values set to /home/root/, but the empty string. Are you able to reproduce this aspect?

Reply:
empty paths are resolved to `/home/root` at runtime when config is parsed. so that will look fine at build time, but at runtime, it will still throw the error

Reply:
https://github.com/lttng/lttng-tools/blob/v2.13.9/src/bin/lttng-sessiond/sessiond-config.c#L475